packages/core/src/components/OverflowTooltip/OverflowTooltip.tsx

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Tooltip, TooltipProps } from '@material-ui/core';
+import React, { useState } from 'react';
+import TextTruncate, { TextTruncateProps } from 'react-text-truncate';
+
+type Props = {
+  text: TextTruncateProps['text'];
+  line?: TextTruncateProps['line'];
+  element?: TextTruncateProps['element'];
+  title?: TooltipProps['title'];
+  placement?: TooltipProps['placement'];
+};
+
+export const OverflowTooltip = (props: Props) => {
+  const [hover, setHover] = useState(false);
+
+  const handleToggled = (truncated: boolean) => {
+    const hoverStatus = truncated ? true : false;
+    setHover(hoverStatus);
+  };
+
+  return (
+    <Tooltip
+      title={props.title ?? (props.text as string)}
```

Comment:
Should we make `text` always be a string in this is needed here?

Reply:
@Rugvip, this is just a composite component, I exposed the propType of the underneath TextTruncate component, which is optional string, whereas title prop of the MUI Tooltip does not expect undefined.